templates/create-daml-app/daml/Setup.daml

```diff
@@ -1,53 +1,86 @@
 module Setup where
 
-import DA.Foldable (forA_)
 import DA.Optional (fromSomeNote)
+import DA.List
 import qualified DA.Text as T
 import Daml.Script
 
--- | A test user for the create-daml-app network.
+-- | A record containing all the parties that we will use in our script
+data Parties = Parties
+  with
+    alice : Party
+    bob : Party
+    charlies: Party
+    public: Party
+
+{- | A test user for the create-daml-app network.
+   | When a participantName is provided, the testUser will be created/or fetched
+   | on the given participant.
+-}
 data TestUser = TestUser with
   alias : Text
   public : Party
+  participantName: Optional ParticipantName
 
 -- | Create a public party, then create three test users.
-setup : Script ()
+setup : Script Parties
 setup = do
   public <- createPublic
-  let aliases = ["Alice", "Bob", "Charlie"]
-  forA_ aliases $ \alias -> createTestUser $ TestUser alias public
+  let aliases = [("Alice", None), ("Bob", None), ("Charlie", None)]
+  parties <- mapA (\(alias, participantNameOpt) ->
+    createTestUser $ TestUser alias public [ParticipantName name | Some name <- participantNameOpt]) aliases
+  pure $ Parties (head parties) (parties !! 1) (parties !! 2) public
 
 -- | Create a test user.
 createTestUser : TestUser -> Script Party
 createTestUser TestUser{alias, public} = do
-  u <- getOrCreateUser alias (Some public)
+  u <- getOrCreateUser alias (Some public) None
```

Comment:
maybe consider defining `defaultParticipant = None` and use that everywhere?

Reply:
yes!